src/evaluation/evaluator.h

```diff
@@ -614,43 +572,428 @@ class Evaluator {
+            if (board.fullMoves > 0) {
+                const auto now = std::chrono::system_clock::now();
+                const auto timeLeft = m_endTime - now;
+                const auto timeSpent = now - m_startTime;
+
+                /* factor is "little less than half" meaning that we give juuust about half the time we spent to search a new depth
+                 * might need tweaking - will do when game phases are implemented */
+                const auto timeLimit = timeSpent / 1.9;
+
+                /* uncommment for debugging */
+
+                /* m_logger.log("d: {}, timeLeft: {}, timeSpent: {}, timeLimit: {}", d, */
+                /*     std::chrono::duration_cast<std::chrono::milliseconds>(timeLeft), */
+                /*     std::chrono::duration_cast<std::chrono::milliseconds>(timeSpent), */
+                /*     std::chrono::duration_cast<std::chrono::milliseconds>(timeLimit)); */
+
+                if (timeLeft < timeLimit) {
+                    /* m_logger.log("Stopped early; saved: {}", std::chrono::duration_cast<std::chrono::milliseconds>(timeLeft)); */
+                    break;
+                }
+            }
+
+            /* Thread voting: https://www.chessprogramming.org/Lazy_SMP */
+            std::array<SearcherResult, s_maxThreads> searchResults {};
+            uint8_t numSearchResults {};
+
+            for (auto& searcher : m_searchers) {
+                searcher->startSearch(d, board, m_threadPool, alpha, beta);
+            }
+
+            for (auto& searcher : m_searchers) {
+                const auto& result = searcher->getSearchResult();
+                if (result.has_value()) {
+                    // If didn't fall out of window and wasn't an immediate early termination, push back to results
+                    if ((result.value().score > alpha) && (result.value().score < beta) && result.value().searchedDepth > 0) {
+                        searchResults.at(numSearchResults) = result.value();
+                        ++numSearchResults;
+                    }
+                }
+            }
+
+            if (numSearchResults == 0) {
+                alpha = s_minScore;
+                beta = s_maxScore;
+
+                continue;
+            }
+
+            m_movesVotes.clear();
+
+            for (uint8_t i = 0; i < numSearchResults; i++) {
+
+                const auto& result = searchResults.at(i);
+
+                const int32_t score = result.score;
+                const uint8_t depth = result.searchedDepth;
+                const auto move = result.pvMove;
+
+                /* Can be tweaked for optimization */
+                int64_t voteWeight = (score - s_minScore) * depth;
+
+                if (!m_movesVotes.count(move)) {
+                    m_movesVotes.insert(move, voteWeight);
+                } else {
+                    m_movesVotes.at(move) += voteWeight;
+                }
```

Comment:
Can be optimized with ["insertOrAssign"](https://en.cppreference.com/w/cpp/container/map/insert_or_assign) :)